pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -97,10 +97,90 @@ var removedPluginsByVersion = []removedPlugins{
 	},
 	{
 		schemeGroupVersion: v1beta2.SchemeGroupVersion.String(),
-		plugins:            []string{"NodeLabel", "ServiceAffinity", "NodePreferAvoidPods"},
+		plugins: []string{
+			"NodeLabel",
+			"ServiceAffinity",
+			"NodePreferAvoidPods",
+			"NodeResourcesLeastAllocated",
+			"NodeResourcesMostAllocated",
+			"RequestedToCapacityRatio",
+		},
 	},
 }
 
+type conflictPlugin struct {
+	ifExist      string
+	conflictWith []string
+
+	// stages defines which stage in scheduler framework, is conflict with other plugin
+	stages []string
+}
+
+type conflictPlugins struct {
+	schemeGroupVersion string
+	plugins            map[string]conflictPlugin
+}
+
+// conflictPluginsByVersion maintains a map of conflict plugins in each version.
+// Remember to add an entry to that list when creating a new component config
+// version (even if the list of conflict plugins is empty).
+var conflictPluginsByVersion = []conflictPlugins{
```

Comment:
change this to a map: 

```
var conflictPluginsByVersion = map[string]map[string]sets.String{
	{
		v1beta1.SchemeGroupVersion.String(): map[string]sets.String{
			"NodeResourcesFit": sets.NewString("NodeResourcesLeastAllocated", "NodeResourcesMostAllocated", "RequestedToCapacityRatio"),
		},
	},
	{
		v1beta2.SchemeGroupVersion.String(): nil,
	},
}
```

you will need to `import "k8s.io/apimachinery/pkg/util/sets"`